program/src/processor/initialize_config.rs

```diff
@@ -1,21 +1,137 @@
+    extension::{transfer_hook::TransferHook, BaseStateWithExtensions, PodStateWithExtensions},
+    pod::{PodAccount, PodMint},
+};
 
-use crate::instruction::accounts::{Context, InitializeConfigAccounts};
+use crate::{
+    err,
+    error::StakeError,
+    instruction::accounts::{Context, InitializeConfigAccounts},
+    processor::REWARDS_PROGRAM_ID,
+    require,
+    state::Config,
+};
 
 /// Creates Stake config account which controls staking parameters.
 ///
 /// ### Accounts:
 ///
 ///   0. `[w]` config
-///   1. `[]` authority
-///   2. `[]` slash_authority
+///   1. `[]` slash_authority
+///   2. `[]` config_authority
 ///   3. `[]` mint
-///   4. `[]` vault_token
+///   4. `[]` vault
 pub fn process_initialize_config(
-    _program_id: &Pubkey,
-    _ctx: Context<InitializeConfigAccounts>,
-    _cooldown_time_seconds: u64,
-    _max_deactivation_basis_points: u16,
+    program_id: &Pubkey,
+    ctx: Context<InitializeConfigAccounts>,
+    cooldown_time_seconds: u64,
+    max_deactivation_basis_points: u16,
 ) -> ProgramResult {
+    // Accounts validation.
+
+    // 1. mint
+    // - owner must the spl token 2022
+    // - must be initialized
+    // - have rewards transfer hook
+
+    require!(
+        ctx.accounts.mint.owner == &spl_token_2022::ID,
+        ProgramError::InvalidAccountOwner,
+        "mint"
+    );
+
+    let mint_data = ctx.accounts.mint.try_borrow_data()?;
+    // unpack checks if the mint is initialized
+    let mint = PodStateWithExtensions::<PodMint>::unpack(&mint_data)?;
+
+    // ensure the mint is configured with the expected `TransferHook` extension
+    if let Ok(transfer_hook) = mint.get_extension::<TransferHook>() {
+        let hook_program_id: Option<Pubkey> = transfer_hook.program_id.into();
+
+        require!(
+            hook_program_id == Some(REWARDS_PROGRAM_ID),
+            StakeError::InvalidTransferHookProgramId,
+            "expected {}, found {:?}",
+            program_id,
+            hook_program_id
+        );
+    } else {
+        return err!(StakeError::MissingTransferHook);
+    }
+
+    // 2. vault (token account)
+    // - must be initialized
+    // - have the vault signer (PDA) as owner
+    // - have the correct mint
+    // - amount equal to 0
+
+    let vault_data = ctx.accounts.vault.try_borrow_data()?;
+    // unpack checks if the token is initialized
+    let vault = PodStateWithExtensions::<PodAccount>::unpack(&vault_data)?;
+
+    let (vault_signer, bump) = Pubkey::find_program_address(
+        &["token-owner".as_bytes(), ctx.accounts.config.key.as_ref()],
+        program_id,
+    );
+
+    require!(
+        vault.base.owner == vault_signer,
+        StakeError::InvalidTokenOwner,
+        "vault"
+    );
+
+    require!(
+        &vault.base.mint == ctx.accounts.mint.key,
+        StakeError::InvalidMint,
+        "vault"
+    );
+
+    let amount: u64 = vault.base.amount.into();
+    require!(amount == 0, StakeError::AmountGreaterThanZero, "vault");
```

Comment:
For this account, can you also make sure that there's no close authority, no delegate, and that the only possible extensions are "immutable owner" and "transfer hook account"? Anything else should be rejected

Reply:
Done! The code enforces that both `ImmutableOwner` and `TransferHookAccount` are present. When either is missing or any other one is found, it fails.